python/tvm/topi/riscv_cpu/tensor_intrin.py

```diff
@@ -0,0 +1,217 @@
+    kernel = te.placeholder((int32_lanes, num_int8_elements), dtype="int8", name="kernel")
+    k = te.reduce_axis((0, num_int8_elements), name="k")
+    C = te.compute(
+        (int32_lanes,),
+        lambda i: te.sum(data[k].astype("int32") * kernel[i, k].astype("int32"), axis=k),
+        name="C",
+    )
+
+    a_buffer = tvm.tir.decl_buffer(
+        data.shape, dtype="uint8", name="a_buffer", offset_factor=1, strides=[1]
+    )
+    b_buffer = tvm.tir.decl_buffer(
+        kernel.shape, dtype="int8", name="b_buffer", offset_factor=1, strides=[te.var("ldw"), 1]
+    )
+
+    def _intrin_func(ins, outs):
+        def _body():
+            ib = tvm.tir.ir_builder.create()
+            ib.emit(
+                tvm.tir.call_extern(
+                    outs[0].dtype,
+                    "dot_uint8_int8_int32_body",
+                    ins[0].access_ptr("r"),
+                    ins[1].access_ptr("r"),
+                    outs[0].access_ptr("w"),
+                )
+            )
+            return ib.get()
+
+        def _reset():
+            ib = tvm.tir.ir_builder.create()
+            ib.emit(
+                tvm.tir.call_extern(
+                    outs[0].dtype, "dot_uint8_int8_int32_reset", outs[0].access_ptr("w")
+                )
+            )
+            return ib.get()
+
+        def _reduce_update():
+            ib = tvm.tir.ir_builder.create()
+            ib.emit(
+                tvm.tir.call_extern(
+                    outs[0].dtype,
+                    "dot_uint8_int8_int32_update",
+                    ins[0].access_ptr("r"),
+                    ins[1].access_ptr("r"),
+                    outs[0].access_ptr("w"),
+                )
+            )
+            return ib.get()
+
+        return _body(), _reset(), _reduce_update()
+
+    buffer_params = {"offset_factor": 1}
+    return te.decl_tensor_intrin(
+        C.op,
+        _intrin_func,
+        binds={data: a_buffer, kernel: b_buffer},
+        default_buffer_params=buffer_params,
+    )
+
+
+def int8_conv2d_impl():
+    """Emit C code for dot product impl."""
+    cc_code = """
+#ifndef TVM_RISCV_CONV2D_INT8
+#define TVM_RISCV_CONV2D_INT8
+#include <riscv_vector.h>
+#include <stdint.h>
+
+#ifdef __cplusplus
+extern "C"
+#endif
+int32_t dot_uint8_int8_int32_reset(int32_t* res) {{
+  // In this function, we set all values in the output array to 0.
+  for (uint32_t i = 0; i < 4; ++i)
+    res[i] = 0;
+  return 0;
+}}
+
+// In this function we need to multiply two vectors, then in the resulting vector calculate the sum of
+// all its elements and store it to output array.
+//
+// Example without vectorization:
+// for (unsigned i = 0; i < 4; ++i) {{
+//   output[i] = 0;
+//   for (unsigned j = 0; j < 4; ++j) {{
+//     output[i] += data[k] * kernel[i][j];
+//   }}
+// }}
+
+#ifdef __cplusplus
+extern "C"
+#endif
+int32_t dot_uint8_int8_int32_body(uint8_t* data, int8_t* kernel, int32_t* output) {{
```

Comment:
Do you by any chance have some benchmark results on how this performs compared to the RVV autovectorizer which is provided by RISCV LLVM/GCC?